### PR TITLE
Fix all miswritten getAdminLink calls in templates

### LINKED ISSUE #11485: Cannot edit the position of a module
**Describe the bug**
In the BO => Design => Positions page, I try to click on the "EDit" button to edit the position of a module nothing happens, the page is just refreshed.

**To Reproduce**
1. Go to BO => Design => Positions page
2. Click on 'Edit' button for any module
4. See the error, the page is refreshed

**Screenshots**
https://drive.google.com/file/d/17ZREr84JEpd6P901usOrpNXJHwRnH75D/view

**Additionnal information**
PrestaShop version: 1.7.5.x
PHP version: 7.1

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.5.x
| Description?  | There are a lot of templates where `getAdminLink` is miswritten, it uses only the controller as a parameter and appends manually the parameter afterwards. This PR fixes every templates so that they are correctly written using full `getAdminLink` parameters. **The same work will have to be done in the controller classes.** It especially fixes the edition of module position, but lots of other links have been improved as well.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #11485
| How to test?  | The most important test is to test module position edition. But lots of other links have been modified and should be tested. The full list is after:

- Test edit position
- Test header:
    - Test Add quick access
    - Test Profile avatar link (right click the user avatar -> Open link) (on legacy page)
    - Test Profile (in submenu -> click on "Your profile") (on legacy page AND migrated pages)
    - Test Logout (in submenu -> click on "Logout") (on legacy page AND migrated pages)
    - Test in notifications orders when empty, one of the three random texts contains a link to "abandonned carts"
- Test Carriers list click on the row links to the edition
- Test Cart view
    - Link to Customer on his name
    - Link to order in "Order Information" (when order is present)
    - Link to Create order in "Order information" (when no order and cart has en customer assigned)
    - Link to Discount (when present)
- Test Customer Service View
    - Modal to forward discussion
    - Buttons Mark as pending, Mark as handled, ...
    - Link to the customer on avatar
    - Link to the customer on his name
    - Form to reply a message
- Test brand view
    - Test link to edit address
- Test Order view
    - Test "View invoice", "View delivery slip" buttons
    - Test PDF documents links (delivery slip, invoice, order slip)
    - Test modal to edit shipping details
    - Test Customer service link (when messages are associated)
    - Test Previous/Next order buttons
    - In history test "Resend mail" button
    - Test change shipping address select form
- Test Order list
    - Test PDF buttons (invoice, delievry slip)
- Test Create order
    - Test search customer card result Details link
    - Test "Add new customer button"
    - After you choose a customer
        - In orders tab test Details link
        - Test the "Add new voucher" button
        - In addresses panel the two edit button
        - Test the "Add a new address button"
    - Finally create or use a cart, and test that the "Create the order" button works
- Test Customers > Outstanding list invoice button
- Test Multistore list "Click here to set a URL for this shop." link
- Test Credit slips list -> "Download credit slip" button
- Themes & Catalog
    - Test "Use this theme" link
    - Test "Choose layouts" button
    - Test "Reset to defaults" button
    - Test "Delete this theme" link
- Customer addresses
    - Test import link
    - Test "Add new address" button
- Attributes & Features
    - Test in list actions "ADD NEW ATTRIBUTES", and "IMPORT" link
    - Test in view list "ADD NEW VALUES" button and "Back to list" link
- Shipping > Carriers    
    - Test link on rows, edit links, delete links

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11531)

### CODE REVIEW

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig [hunk 1]
```diff
@@ -154,7 +154,16 @@
...
                         <div class="module-column-actions">
                           <div class="btn-group">
-                            <a class="btn btn-outline-primary btn-sm" href="{{ getAdminLink('AdminModulesPositions') }}&amp;id_module={{ moduleInstance.id }}&amp;id_hook={{ hook['id_hook'] }}&amp;editGraft{% if selectedModule %}&amp;show_modules={{ selectedModule }}{% endif %}">
+                            {% set linkParams = {
+                              'id_module': moduleInstance.id,
+                              'id_hook': hook['id_hook'],
+                              'editGraft': 1
+                            } %}
+                            {% if selectedModule %}
+                              {% set linkParams = liParams|merge({'show_modules': selectedModule}) %}
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig]: `linkParams` and not `LiParams`?

File: classes/controller/AdminController.php [hunk 2]
```diff
@@ -1966,7 +1967,7 @@ private function getNotificationTip($type)
                         array(
                             '[1]' => '<strong>',
                             '[/1]' => '</strong>',
-                            '[2]' => '<a href="' . $this->context->link->getAdminLink('AdminCarts') . '&action=filterOnlyAbandonedCarts">',
+                            '[2]' => '<a href="' . $this->context->link->getAdminLink('AdminCarts', true, array(), array('action' => 'filterOnlyAbandonedCarts')) . '">',
```
Comment (PierreRambaud) [inline classes/controller/AdminController.php:1970]: Why `array()` and not `[]` like above?

File: admin-dev/themes/new-theme/template/components/layout/employee_dropdown.tpl [hunk 3]
```diff
@@ -32,11 +32,11 @@
       <img class="avatar rounded-circle" src="{$employee->getImage()}" />
       <span>{$employee->firstname} {$employee->lastname}</span>
     </div>
-    <a class="dropdown-item employee-link profile-link" href="{$link->getAdminLink('AdminEmployees')|escape:'html':'UTF-8'}&amp;id_employee={$employee->id|intval}&amp;updateemployee">
+    <a class="dropdown-item employee-link profile-link" href="{$link->getAdminLink('AdminEmployees', true, [], ['id_employee' => $employee->id|intval, 'updateemployee' => 1])|escape:'html':'UTF-8'}">
       <i class="material-icons">settings_applications</i>
       {l s='Your profile'}
     </a>
-    <a class="dropdown-item employee-link" id="header_logout" href="{$login_link|escape:'html':'UTF-8'}&amp;logout">
+    <a class="dropdown-item employee-link" id="header_logout" href="{$logout_link|escape:'html':'UTF-8'}">
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/template/components/layout/employee_dropdown.tpl:39]: :+1:

**Comment by Quetzacoalt91** [comment]
Nothing to add regarding the PR content. However, if you had to update all URLs generated, you should probably warn the module developers to do the same thing in https://devdocs.prestashop.com/1.7/modules/core_updates/1.7.5/

If their module cannot reach specific pages of the back office and they are not aware of it, this is bad.

**Comment by jolelievre** [comment]
@Quetzacoalt91 I mentioned it in the routing section https://devdocs.prestashop.com/1.7/development/architecture/migration-guide/controller-routing/#more-about-the-legacy-link-property
Do you think it's worth repeating it the the 1.7.5 changes? with a link to the routing page maybe?

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig (hunk 1 above)
Comment (jolelievre) [reply src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig]: thanks!

**Comment by Quetzacoalt91** [comment]
Nice to see you already wrote something about this. But yes, the link I gave you is mentioned in the Build article, to help contributors knowing what are the potential regression brought by 1.7.5.0. If we can add a link to the other page, that'd be great.

File: classes/controller/AdminController.php (hunk 2 above)
Comment (jolelievre) [reply classes/controller/AdminController.php:1970]: this class uses the `array()` notation everywhere so I prefer to write it this way

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig (hunk 1 above)
Comment (jolelievre) [reply src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig]: @PierreRambaud done!

**Comment by jolelievre** [comment]
@Quetzacoalt91 here is the update for dev doc: https://github.com/PrestaShop/docs/pull/168

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
Thank you @jolelievre

**Pull request merged**